src/GridFieldOrderableRows.php

```diff
@@ -648,10 +715,13 @@ protected function getSortTableClauseForIds(DataList $list, $ids)
             $value = '= ' . (int) $ids;
         }
 
-        if ($list instanceof ManyManyList) {
-            $extra = $list->getExtraFields();
-            $key   = $list->getLocalKey();
-            $foreignKey = $list->getForeignKey();
+        if ($this->isManyMany($list)) {
+            $intropector = $this->getManyManyInspector($list);
+            $extra = $list instanceof ManyManyList ?
+                $intropector->getExtraFields() :
+                DataObjectSchema::create()->fieldSpecs($intropector->getJoinClass(), DataObjectSchema::DB_ONLY);
```

Comment:
Multi line ternary isn't the best for readability, perhaps you could make this an if statement? Failing that, putting the `?` and `:` at the start of lines helps to indicate that it's part of a ternary condition and not just bad indentation